internal/primitives/database/mem.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package database
+
+import (
+	"sync"
+
+	"github.com/ChainSafe/gossamer/internal/primitives/core/hash"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime"
+)
+
+type countValue struct {
```

Comment:
what does this represent? I think a comment here could be useful

Reply:
🤔 it's used to store reference counting and the value?

Reply:
Seeing count and jumping to this being a reference count, for a type with a name like `countValue` is not intuitive to me at least, especially in go, which is why I thought it could be useful. However, this is a nit/so if you don't think its helpful can leave as is

Reply:
I don't remember where I read this, but it's usually a best practice to name your helper tuple like structs as just the concatenation of the attribute names.  I'm not opposed to changing it to `refCountValue` though.

Reply:
If this is best practice I don't mind leaving as is, totally up to you. I didn't know that about he naming of a helper duple so good learning point for me